### ISSUE TITLE
Issue when using --keep-ambig with --base-maf

### ISSUE THREAD

**Issue by aholleman** [choishingwan/PRSice#122]
Hi,

I posted this on the Google group but wanted to make sure it didn't lost in the back and forth about a separate issue.

I've come across an issue related to --keep-ambig. I'm using the latest PRSice release (v2.2.3). When I use --keep-ambig along with --base-maf, the number of variants ultimately included from the base file is equal to the number of non-ambiguous variants left after the --base-maf filter plus the total number of ambiguous variants (without any MAF filtering). I've included the relevant section of the log file, below. PRSice correctly identifies that 2,966,342 variants (including both ambiguous and non-ambiguous variants) out of the total 3.8mil variants should be excluded due to MAF, but then it ends up actually applying the exclusion only to the non-ambiguous variants, resulting in 731,647 non-ambig vars plus all 580,875 ambig vars = 1,312,522 vars. Is there a way to have PRSice apply the MAF exclusion to the ambiguous variants as well as the non-ambig vars?

Thanks,
Aaron

Reading 100.00%
Base file: 
CHOP_GWAS/CHOP_CTD_casecontrol_1KG.in_3826927SNPs_withAmbig_withMAF.txt 
3826927 variant(s) observed in base file, with: 
580875 ambiguous variant(s) 
1 NA stat/p-value observed 
2966342 variant(s) excluded due to MAF threshold 
1312522 total variant(s) included from base file

**Comment by choishingwan** [comment]
`--base-maf` occurs before ambiguous SNP filtering, as a result of that, the numebr of variants excluded due to MAF threshold should be equalt to the total number of SNPs filtered (included ambiguous SNPs). If you don't mind, could you please have a look at the number and see if that's the case? 
Thanks

**Comment by aholleman** [comment]
Hi Sam,


So when I use --keep-ambig, the number reported as being filtered out due to --base-maf is correct (it includes both ambig and non-ambig SNPs), but then it appears that all of the ambig vars (including those that should be filtered out based on MAF) are added back in to the base file. I've reproduce full log files below, the first is for a run with --keep-ambig and the second is for an identical run without --keep-ambig.



WITH --KEEP-AMBIG


PRSice 2.2.3 (15 July 2019)

https://github.com/choishingwan/PRSice

(C) 2016-2017 Shing Wan (Sam) Choi, Jack Euesden, Cathryn M. Lewis, Paul F. O'Reilly

GNU General Public License v3


If you use PRSice in any published work, please cite:

Jack Euesden Cathryn M. Lewis Paul F. O'Reilly (2015)

PRSice: Polygenic Risk Score software.

Bioinformatics 31 (9): 1466-1468


2019-07-15 16:50:32

./PRSice_mac_v2.2.3 \

    --A1 EA \

    --A2 NEA \

    --bar-levels 1e-06,1e-05,0.0001,0.001,0.005,0.01,0.05,0.1,0.2,0.3,0.4,0.5,1 \

    --base CHOP_GWAS/CHOP_CTD_casecontrol_1KG.in_3826927SNPs_withAmbig_withMAF.txt \

    --base-maf MAF.hrc.exclude1000G,0.35 \

    --binary-target T \

    --bp POS \

    --chr CHR \

    --clump-kb 250 \

    --clump-p 1.000000 \

    --clump-r2 0.100000 \

    --cov-col Sex \

    --cov-file Covariates_DS_Imputed_403_fam.20PCs.txt \

    --fastscore  \

    --keep-ambig  \

    --lower 1e-06 \

    --missing MEAN_IMPUTE \

    --model add \

    --or  \

    --out PRSice_Results/DSAffyImputed_403_CHOP.casecontrol_maf35_geno02_noMHC_Sex_withAmbig_v2.2.3 \

    --pvalue Corr_Trend__P \

    --score std \

    --seed 2034760375 \

    --snp SNP \

    --stat OR \

    --target DS_Affy_postImputation_403_geno02_maf01target_noMHC \

    --thread 1 \

    --upper 1



Initializing Genotype file:

DS_Affy_postImputation_403_geno02_maf01target_noMHC (bed)




Start processing

CHOP_CTD_casecontrol_1KG.in_3826927SNPs_withAmbig_withMAF

==================================================


Base file:

CHOP_GWAS/CHOP_CTD_casecontrol_1KG.in_3826927SNPs_withAmbig_withMAF.txt

3826927 variant(s) observed in base file, with:

580875 ambiguous variant(s)

1 NA stat/p-value observed

2966342 variant(s) excluded due to MAF threshold

1312522 total variant(s) included from base file **This includes all 580875 ambiguous vars rather than only including those with MAF above the threshold






Loading Genotype info from target

==================================================


403 people (209 male(s), 194 female(s)) observed

403 founder(s) included


580875 ambiguous variant(s) kept

1312522 variant(s) included




Start processing gene set information

==================================================


1 region included


There are a total of 1 phenotype to process




Start performing clumping


16384 MB RAM detected; reserving 1 MB for clumping

Allocated 1 MB successfully


Number of variant(s) after clumping : 62183


Phenotype is a binary phenotype

212 control(s)

191 case(s)




Include Covariates:

Name Missing Number of levels

Sex 0 -




Processing the covariate file:

Covariates_DS_Imputed_403_fam.20PCs.txt

==============================




After reading the covariate file, 403 sample(s) included in

the analysis




There are 1 region(s) with p-value between 0.1 and 1e-5

([1;31mmay not be significant[0m);






WITHOUT --KEEP-AMBIG


PRSice 2.2.3 (15 July 2019)

https://github.com/choishingwan/PRSice

(C) 2016-2017 Shing Wan (Sam) Choi, Jack Euesden, Cathryn M. Lewis, Paul F. O'Reilly

GNU General Public License v3


If you use PRSice in any published work, please cite:

Jack Euesden Cathryn M. Lewis Paul F. O'Reilly (2015)

PRSice: Polygenic Risk Score software.

Bioinformatics 31 (9): 1466-1468


2019-07-16 16:15:56

./PRSice_mac_v2.2.3 \

    --A1 EA \

    --A2 NEA \

    --bar-levels 1e-06,1e-05,0.0001,0.001,0.005,0.01,0.05,0.1,0.2,0.3,0.4,0.5,1 \

    --base CHOP_GWAS/CHOP_CTD_casecontrol_1KG.in_3826927SNPs_withAmbig_withMAF.txt \

    --base-maf MAF.hrc.exclude1000G,0.35 \

    --binary-target T \

    --bp POS \

    --chr CHR \

    --clump-kb 250 \

    --clump-p 1.000000 \

    --clump-r2 0.100000 \

    --cov-col Sex \

    --cov-file Covariates_DS_Imputed_403_fam.20PCs.txt \

    --fastscore  \

    --lower 1e-06 \

    --missing MEAN_IMPUTE \

    --model add \

    --or  \

    --out PRSice_Results/DSAffyImputed_403_CHOP.casecontrol_maf35_geno02_noMHC_Sex_withoutAmbig_v2.2.3 \

    --pvalue Corr_Trend__P \

    --score std \

    --seed 3875810211 \

    --snp SNP \

    --stat OR \

    --target DS_Affy_postImputation_403_geno02_maf01target_noMHC \

    --thread 1 \

    --upper 1



Initializing Genotype file:

DS_Affy_postImputation_403_geno02_maf01target_noMHC (bed)




Start processing

CHOP_CTD_casecontrol_1KG.in_3826927SNPs_withAmbig_withMAF

==================================================


Base file:

CHOP_GWAS/CHOP_CTD_casecontrol_1KG.in_3826927SNPs_withAmbig_withMAF.txt

3826927 variant(s) observed in base file, with:

580875 ambiguous variant(s) excluded

1 NA stat/p-value observed

2966342 variant(s) excluded due to MAF threshold

731647 total variant(s) included from base file **This number 731647 plus the 580875 ambig vars equals the total # included from base file in the above log that used --keep-ambig (1,312,522 vars)






Loading Genotype info from target

==================================================


403 people (209 male(s), 194 female(s)) observed

403 founder(s) included


731647 variant(s) included




Start processing gene set information

==================================================


1 region included


There are a total of 1 phenotype to process




Start performing clumping


16384 MB RAM detected; reserving 1 MB for clumping

Allocated 1 MB successfully


Number of variant(s) after clumping : 27513


Phenotype is a binary phenotype

212 control(s)

191 case(s)




Include Covariates:

Name Missing Number of levels

Sex 0 -




Processing the covariate file:

Covariates_DS_Imputed_403_fam.20PCs.txt

==============================




After reading the covariate file, 403 sample(s) included in

the analysis




There are 1 region(s)/phenotype(s) with p-value > 0.1

([1;31mnot significant[0m);




________________________________

**Comment by choishingwan** [comment]
Thanks, this help me to pinpoint the problem. 
Due to some careless mistake, the `--keep-ambig` flag actually reset our filtering flag. Therefore whenever `--keep-ambig` is defined, none-of the filtering will be performed. 

The commit #8dc4483fee76e7f6017270f70fef35be5788ef5b should now work. Unfortauntely, I am currently busy on something else. Maybe you can compile the software yourself.
```
g++  -std=c++11 -O3 -march=native -I inc/ -isystem lib/ src/*.cpp -lpthread -lz -o PRSice_mac
```

**Comment by aholleman** [comment]
I don't think I have the skills to compile it myself, but it's not a problem. I can do some manual filtering which should accomplish my goals.

Thanks,
Aaron

**Comment by choishingwan** [comment]
I still have one additional issues to tackle at the moment. Once that's
solved, I will push out 2.2.4, which should contain the fix.
Dr Shing Wan Choi
Postdoctoral Fellow
Genetics and Genomic Sciences
Icahn School of Medicine, Mount Sinai, NYC

**Comment by aholleman** [comment]
Sounds great. Thanks so much!

________________________________
